src/dataExplorer/components/Schema.tsx

```diff
@@ -35,7 +36,7 @@ const FieldsTags: FC = () => {
 
   useEffect(() => {
     setSearchTerm('')
```

Comment:
Generally it's not a great practice to disable `exhaustive deps`, because the fact that we're getting a warning means that our `useEffect` hook isn't behaving as intended: updating only those portions of the DOM on which the useEffect hook actually relies. Here's the longer React thread on this: https://github.com/facebook/react/issues/14920#issuecomment-471070149

A way to implement this without creating useEffect issues could be to just add a `setSearchTerm` to the click handler.

Reply:
Yeah, I agree @helenosheaa - if it's going to be a bigger issue to work around it, disabling the line is okay. It just gives me some pause because I've seen render issues on some pages where this is used, and there seems to be one here too.

For context - a ways back we had a ticket to get rid of these `eslint-disable-line` references throughout the codebase to try to reduce some code smell, but due to priorities we've dropped it for now.

Reply:
I just read the link that @wdoconnell shared above. It was helpful! I did see somewhere in code that the re-render wasn't as expected and I wonder if the behavior is related to the useEffect dependency list. I'll try to remove `eslint-disable-line` and eslint warnings that's introduced in this PR as much as possible

Reply:
@wdoconnell @helenosheaa  I pushed two commits: 
* https://github.com/influxdata/ui/pull/6648/commits/f9752753e11ee915b7b279f9dfa8dda38658081e removes the `eslint-disable` comment by completing the dependency list and 
* https://github.com/influxdata/ui/pull/6648/commits/8678c83055c2b55615b395bbafb1d0844c8b432c adds comments why `eslint-disable` is there